pygdf/numerical.py

```diff
@@ -233,7 +233,24 @@ def default_na_value(self):
             raise TypeError(
                 "numeric column of {} has no NaN value".format(self.dtype))
 
-    def join(self, other, how='left', return_indexers=False):
+    def join(self, other, how='left', return_indexers=False, type='sort'):
+
+        # Single column join using sort-based implementation
+        if type == 'sort':
+            return self._sortjoin(other=other, how=how, return_indexers=return_indexers)
+        elif type == 'hash':
+            # Get list of columns from self with left_on and from other with right_on
+            return self._hashjoin(other=other, left_on=left_on, right_on=right_on, how=how, return_indexers=return_indexers)
+        else:
+            raise ValueError('Unsupported join type')
+
+    def _hashjoin(self, other, how='left', return_indexers=False):
+        msg = "numerical hash join not implemented yet."
```

Comment:
This error message doesn't seem right. Is this for joining on the index? If so lets update the error message to "Hash joins on index not implemented yet."